Source/WebCore/platform/text/mac/TextBoundaries.mm

```diff
@@ -191,32 +233,47 @@ void findWordBoundary(StringView text, int position, int* start, int* end)
         *end = 0;
         return;
     }
-
-    unsigned pos = position;
-    if (pos >= text.length() && pos)
-        pos = text.length() - 1;
+    if (static_cast<unsigned>(position) >= text.length()) {
+        // This makes no sense. Return what we got in.
+        ASSERT(true);
+        *start = position;
+        *end = position;
+        return;
+    }
```

Comment:
I'm not sure I follow the above. Instead of returning the end of the text if the position is beyond it, you think we should just return the original position? I think both are valid, but I worry that places might be depending on this behavior, but I haven't investigated. I'm also not sure why you assert(true)?

Reply:
I'm asserting because calling this function with a position beyond the length of the string is, afaict, complete nonsense (and would have crashed before this change). But in order to avoid a crash, we need to return something so rather than asserting the inequality twice I included ASSERT(true) in the `if` statement.